build-tools/plans/command-conversion-to-vnext.md

```diff
@@ -0,0 +1,280 @@
+# Plan: Converting Commands to VNext Style
+
+This document provides guidance on converting commands from the legacy `PackageCommand` pattern to the newer
+`BaseCommandWithBuildProject` pattern (vnext style) that uses the APIs in the `@fluid-tools/build-infrastructure`
+package.
+
+## Background
+
+The build-tools project has two styles of building commands that operate on packages:
+
+1. **Legacy Style (PackageCommand)**: Uses the `Context` class from `@fluidframework/build-tools` and the
+   `PackageCommand` base class defined in `BasePackageCommand.ts`. This style relies on the older package
+   infrastructure.
+
+2. **VNext Style (BaseCommandWithBuildProject)**: Uses the `IBuildProject` and related types from
+   `@fluid-tools/build-infrastructure`. This newer pattern provides better typing, more flexible filtering,
+   and a cleaner separation of concerns.
+
+## Key Differences
+
+### Base Classes
+
+| Aspect | Legacy (PackageCommand) | VNext (BaseCommandWithBuildProject) |
+|--------|------------------------|-------------------------------------|
+| Base Class | `PackageCommand<T>` | `BaseCommandWithBuildProject<T>` |
+| Package Type | `Package` from `@fluidframework/build-tools` | `IPackage` from `@fluid-tools/build-infrastructure` |
+| Context | `Context` from `library/context.js` | `IBuildProject` from `build-infrastructure` |
+| Get Context | `this.getContext()` | `this.getBuildProject()` |
+
+### Package Selection and Filtering
+
+| Aspect | Legacy | VNext |
+|--------|--------|-------|
+| Selection | `selectionFlags` defined in `flags.ts` | Can use `build-infrastructure` filter types |
+| Filter | `filterFlags` defined in `flags.ts` | `PackageFilterOptions` from `build-infrastructure` |
+| Default Selection | `defaultSelection` abstract property | Can define custom defaults |
+
+### Processing Packages
+
+| Aspect | Legacy | VNext |
+|--------|--------|-------|
+| Iteration | `processPackage(pkg, kind)` called per package | Manual iteration over packages |
+| Concurrency | Built-in with `async.mapLimit` | Manual implementation |
+| Error Handling | Built-in error collection | Manual implementation |
```

Comment:
We should build the base classes necessary to replicate this as noted elsewhere.